src/engraving/dom/score.cpp

```diff
@@ -884,32 +885,72 @@ void Score::spell()
             for (track_idx_t track = strack; track < etrack; ++track) {
                 EngravingItem* e = s->element(track);
                 if (e && e->type() == ElementType::CHORD) {
-                    std::copy_if(toChord(e)->notes().begin(), toChord(e)->notes().end(),
+                    Chord* c = toChord(e);
+                    std::copy_if(c->notes().begin(), c->notes().end(),
                                  std::back_inserter(notes), [this](EngravingItem* ce) { return selection().isNone() || ce->selected(); });
+                    for (Chord* g : c->graceNotes()) {
+                        std::copy_if(g->notes().begin(), g->notes().end(),
+                                     std::back_inserter(notes),
+                                     [this](EngravingItem* ce) { return selection().isNone() || ce->selected(); });
+                    }
                 }
             }
         }
         spellNotelist(notes);
     }
 }
 
-void Score::spell(staff_idx_t startStaff, staff_idx_t endStaff, Segment* startSegment, Segment* endSegment)
+static std::vector<Note*> allNotesInScore(Score* score)
 {
-    for (staff_idx_t i = startStaff; i < endStaff; ++i) {
-        std::vector<Note*> notes;
-        for (Segment* s = startSegment; s && s != endSegment; s = s->next()) {
+    std::vector<Note*> notes;
+    for (staff_idx_t i = 0; i < score->nstaves(); ++i) {
+        for (Segment* s = score->firstSegment(SegmentType::All); s; s = s->next1()) {
             track_idx_t strack = i * VOICES;
             track_idx_t etrack = strack + VOICES;
             for (track_idx_t track = strack; track < etrack; ++track) {
                 EngravingItem* e = s->element(track);
                 if (e && e->type() == ElementType::CHORD) {
-                    notes.insert(notes.end(),
-                                 toChord(e)->notes().begin(),
-                                 toChord(e)->notes().end());
+                    Chord* c = toChord(e);
+                    notes.insert(notes.end(), c->notes().begin(), c->notes().end());
+                    for (Chord* g : c->graceNotes()) {
+                        notes.insert(notes.end(), g->notes().begin(), g->notes().end());
+                    }
                 }
             }
         }
-        spellNotelist(notes);
+    }
+    return notes;
+}
```

Comment:
I added this in order to implement `spellWithSharps` and `spellWithFlats`, but we could also use it to implement `spell`. The only reason I didn't do that is it would result in calling `spellNotelist` on the whole score at once, while the existing implementation does it one staff at a time. That could result in some subtle behavior changes.